### PR TITLE
Improve menus due to improvements in electron

### PR DESCRIPTION
**Pull request by Razzeee**
Just updated to the newest version of electron and used some of the new features, to streamline our menus.

See each commit for details.

Tested on windows 10 so far.

### CODE REVIEW

**Comment by jasonblais** [comment]
Thanks @Razzeee! Can I have your help to also post the latest artifacts on circleci.com, so we can have others in the community help test the pull request?

**Comment by Razzeee** [comment]
What do you mean? Upstream automatically builds, when you send a pr https://circleci.com/gh/mattermost/desktop/463#artifacts

**Comment by jasonblais** [comment]
Sorry for the confusion, `a link to the corresponding artifact on circleci.com` is what I meant

Thank you for adding it!

**Comment by yuya-oc** [comment]
Just curious, "Toggle Fullscreen" is only localized on OS X. Do you know anything about this?

toggle_fullscreen_osx

**Comment by Razzeee** [comment]
No clue, sorry. Maybe ask in https://github.com/electron/electron/pull/6166

**Comment by Razzeee** [comment]
Rebased, not sure if/how to add it to the changelog

**Comment by jasonblais** [comment]
Thanks again @Razzeee! For the changelog entry, will this have any effect on the user interface, or on performance?

Would we be able to have two community members testing this pull request, one each in OS X and Linux/Ubuntu? We would like to have at least two reviewers before it's merged.

The artifacts to test can be found here: https://circleci.com/gh/mattermost/desktop/505#artifacts

**Comment by Razzeee** [comment]
Well it should be a little bit more platform specifical correct. For example the quit button on Windows is now named Exit as it's for pretty much every other app on Windows. Little things like that.

**Comment by jasonblais** [comment]
Ah cool, thanks @Razzeee  - I'll make a list of changes I've noticed while testing it on Windows, and we can maybe prepare a similar one for OS X and Linux?

Will know better than how to update our docs

**Comment by jasonblais** [comment]
Thanks again @Razzeee - tested on Windows and didn't find issues with functionality.

However, I compared to other messaging platforms such as Slack and Franz, which use "Quit" for quitting the app. I'm not sure if it's obvious whether "Exit" closes or quits the app.

We're also losing a shortcut, which is CTRL+Q to quit the app.

Any thoughts?

**Comment by Razzeee** [comment]
Well that's standard for windows, as I can't think of an app that has a close shortcut on Windows. On osx it's still named quit and has a shortcut. As that's the platform way of doing it. 

Compare the menu to notepad++ or sublime on Windows, not Franz/Slack as that's also a electron app and not native.

**Comment by yuya-oc** [comment]
I think it's not important whether the app is made of Electron.

Generally, Windows apps can be closed by system shortcuts like Alt+F4, so the shortcut is not necessary. But this app keeps the window as hiding and not be closed in order to communicate with the server. So I feel it's not bad idea to keep the shortcut.

**Comment by Razzeee** [comment]
Are you sure that alt + f4 doesn't close the app? Would seem like a bug to me.

**Comment by yuya-oc** [comment]
Not bug, but "not be closed" was not enough to describe the behavior. When closing the main window, the app doesn't close and just hide it in close-event. As the result, the window still exists internally when using Alt+F4. So I wrote "not be closed".

**Comment by Razzeee** [comment]
Got you. So when we want that, we should overwrite the shortcut for "Exit" on windows with Alt + F4

**Comment by yuya-oc** [comment]
I expect Alt+F4 to close a "window" like Ctrl+W. For example, Chrome closes a window on Alt+F4, but the app is still running when there are other windows or background tasks, and Ctrl+Shift+Q is provided to exit. So I feel we should use other shortcuts to finish the app (So I had added Ctrl+Q).
How do you feel?

**Comment by Razzeee** [comment]
I really don't have an opinion about this. In my mind the best parts about letting it be handled by electron is the consistent wording, familiar shortcuts and implementations. And I don't think we make it worse if we add one shortcut to one of the entries.

**Comment by jasonblais** [comment]
Thanks for the thoughtful discussion here!

1) I did a little research and found [Microsoft's Design apps for the Windows desktop guide](https://msdn.microsoft.com/en-us/library/windows/desktop/dn742392.aspx). Here they spell out that **File > Exit** in the menu bar is standard and **File > Quit** doesn't really exist, except in the command line.

To make the Desktop App consistent with other apps on Windows and therefore more obvious to end users, agree @Razzeee to change it to **File > Exit**.

2) However, propose `CRL+Q` would still be used as the shortcut to **Exit** (or CMD+Q to quit on Mac), and `ALT+F4` to **Close** the app to the system tray. This is current behaviour and is similar to other apps when there is a program running in the background (along the lines of what @yuya-oc mentioned above).

@Razzeee If you're fine with 2), can we have your help updating the menu bar so that `CTRL/CMD+Q` is used to quit the app?

**Comment by Razzeee** [comment]
Quit as in shutdown with nothing running in the background? Just to be clear?

**Comment by jasonblais** [comment]
Yes, that would be correct @Razzeee. Thanks for confirming

**Comment by Razzeee** [comment]
@jasonblais 
changed

@yuya-oc 
you might want to redo the review, as I had to add some special handling for the exiting via STRG-Q on windows

**Comment by jasonblais** [comment]
Great, will use this the CircleCI during my work tomorrow, thanks @Razzeee for the updates!

Any OS X or Linux testers who would like to help test it out? https://circleci.com/gh/mattermost/desktop/537#artifacts

**Comment by jasonblais** [comment]
Update: I tested on Windows 10 and at one point the app completely froze for me and I had to quit it.

Might have been a one-off thing and not related to this PR, but I'll try again tomorrow to see if it reproduces. Otherwise no issues spotted

**Comment by jasonblais** [comment]
Haven't reproduced it, so pass from me. No issues found during testing.

@Razzeee Might need one more rebase before we merge it.

@yuya-oc can we have your help reviewing this PR before we merge it?

**Comment by Razzeee** [comment]
Rebased

File: package.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
     "electron-builder": "5.2.1",
     "electron-connect": "^0.3.7",
     "electron-packager": "^7.0.1",
-    "electron-prebuilt": "1.2.6",
+    "electron-prebuilt": "1.3.1",
```
Comment (yuya-oc) [inline package.json]: Please keep v1.2.6 here. v1.3 should be tested in #214.
Please DON'T multiple things in one PR...

File: src/main.js [hunk 2]
```diff
@@ -163,6 +163,10 @@ app.on('activate', function(event) {
 });
 
 app.on('before-quit', function() {
+  // Make sure tray icon gets removed if the user exits via STRG-Q
```
Comment (yuya-oc) [inline src/main.js]: Ctrl-Q?

**Comment by yuya-oc** [comment]
It looks okay except electron version.

File: package.json (hunk 1 above)
Comment (Razzeee) [inline package.json]: 1.2.6 won't work, needs to be 1.2.7 at least.
Sorry, I think I mixed up this PR with the other one.

Comment (Razzeee) [inline package.json]: Nvm, I rebased and it conflicted on the version and I resolved it. Created that false bump due to an error in resolving.

**Comment by Razzeee** [comment]
Fixed your comments, thank you.

**Comment by yuya-oc** [comment]
Tested on OS X 10.11.5. It looks okay. Thanks!

**Comment by jasonblais** [comment]
Thanks @yuya-oc! I think this is ready to merge, has been tested in OS X and Windows.

@Razzeee excellent work here, thank you!!

**Comment by Razzeee** [comment]
There are now new roles for the zoom functions:
https://github.com/electron/electron/pull/6777
Should probably done in a follow up PR.

**Comment by yuya-oc** [comment]
Thanks!

**Pull request merged**